plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -132,98 +114,97 @@ private void setupDataBinding() {
-                                                              bucketObservable,
+    ISWTObservableValue overrideButton = WidgetProperties.selection().observe(overrideDefaultBucketButton);
+    ISWTObservableValue bucketField = WidgetProperties.text(SWT.Modify).observe(bucket);
+    ISWTObservableValue bucketLabelEnablement = WidgetProperties.enabled().observe(bucketLabel);
+    ISWTObservableValue bucketFieldEnablement = WidgetProperties.enabled().observe(bucket);
+
+    // use an intermediary value to control the enabled state of the label and the field based on the override
+    // checkbox's state
+    WritableValue enablement = new WritableValue();
+    context.bindValue(overrideButton, enablement);
+    context.bindValue(bucketLabelEnablement, enablement);
+    context.bindValue(bucketFieldEnablement, enablement);
+
+    IObservableValue overrideModelObservable = PojoProperties.value("overrideDefaultBucket").observe(model);
+    IObservableValue bucketModelObservable = PojoProperties.value("bucket").observe(model);
+
+    context.bindValue(enablement, overrideModelObservable);
+    context.bindValue(bucketField, bucketModelObservable);
+
+    context.addValidationStatusProvider(new OverrideValidator(overrideButton,
+                                                              bucketField,
                                                               new BucketNameValidator()));
   }
 
   @Override
   public boolean performOk() {
     if (isValid()) {
-      try {
-        savePreferences();
-        return true;
-      } catch (IOException exception) {
-        logger.log(Level.SEVERE, "Could not save deploy preferences", exception);
-        MessageDialog.openError(getShell(),
-                                Messages.getString("deploy.preferences.save.error.title"),
-                                Messages.getString("deploy.preferences.save.error.message",
-                                                   exception.getLocalizedMessage()));
-      }
+      return savePreferences();
     }
     return false;
   }
 
-  private void savePreferences() throws IOException {
-    IPreferenceStore preferenceStore = getPreferenceStore();
-
-    preferenceStore.setValue(PREF_PROJECT_ID, model.getProjectId());
-    preferenceStore.setValue(PREF_PROMPT_FOR_PROJECT_ID, model.isPromptForProjectId());
-    preferenceStore.setValue(PREF_OVERRIDE_DEFAULT_VERSIONING, model.isOverrideDefaultVersioning());
-    preferenceStore.setValue(PREF_CUSTOM_VERSION, model.getVersion());
-    preferenceStore.setValue(PREF_ENABLE_AUTO_PROMOTE, model.isAutoPromote());
-    preferenceStore.setValue(PREF_OVERRIDE_DEFAULT_BUCKET, model.isOverrideDefaultBucket());
-    preferenceStore.setValue(PREF_CUSTOM_BUCKET, model.getBucket());
-
-    ScopedPreferenceStore scopedPreferenceStore = (ScopedPreferenceStore) preferenceStore;
-    if (scopedPreferenceStore.needsSaving()) {
-      scopedPreferenceStore.save();
+  private boolean savePreferences() {
```

Comment:
Error return codes are unusual in java. Consider throwing an exception on failure to save instead and handling it up the call stack.